winui3/DispatcherQueueUpdates.md

```diff
@@ -0,0 +1,134 @@
+associated with it. So by default, the .NET Native/Framework sync context works automatically on the UI
+thread of a UWP app, as well as any thread that has a DispatcherQueue running on it
+(a DispatcherQueue can be put onto a thread using the
+[DispatcherQueueController.CreateOnDedicatedThread](https://docs.microsoft.com/uwp/api/Windows.System.DispatcherQueueController.CreateOnDedicatedThread)
+method or [CreateDispatcherQueueController](https://docs.microsoft.com/en-us/windows/win32/api/dispatcherqueue/nf-dispatcherqueue-createdispatcherqueuecontroller)
+function).
+
+In .NET5, however, there is no default SynchronizationContext on every thread, and no built-in 
+understanding of CoreDispatcher or DispatcherQueue. 
+
+This spec adds this functionality for WinUI3 to work with SynchronizationContext, by defining a subclass of it:
+`DispatcherQueueSynchronizationContext`. This can be set as a thread's SynchronizatonContext, and when posted to
+will post to the underlying DispatcherQueue. Note that this is a .NET API, not a WinRT API.
+
+Also described here is a WinRT method to create a DispatcherQueue, providing an alternative to the current COM API.
+
+All of these APIs are for WinUI3/Renuion, not being added to the OS.
+
+# Description
+
+## DispatcherQueueSynchronizationContext
+
+Use the `DispatcherQueueSynchronizationContext` to queue work to another thread, 
+which has an associated [DispatcherQueue](https://docs.microsoft.com/uwp/api/Windows.System.DispatcherQueue).
+See the base class
+[SynchronizationContext](https://docs.microsoft.com/dotnet/api/System.Threading.SynchronizationContext)
+for more information on how to use a SynchronizationContext.
+
+### Remarks
+
+This class is only available to .NET5 or above applications.
+
+The Send method is not supported and throws a NotSupportedException.
+
+## DispatcherQueueController.CreateOnCurrentThread
+
+Creates a [DispatcherQueueController](https://docs.microsoft.com/uwp/api/Windows.System.DispatcherQueueController) 
+on the caller's thread. Use the created DispatcherQueueController to create and manage the 
+lifetime of a [DispatcherQueue](https://docs.microsoft.com/uwp/api/Windows.System.DispatcherQueue)
+to run queued tasks in priority order on the Dispatcher queue's thread.
+
+This method will fail with an error if the current thread already has a DispatcherQueue associated with it.
+You can use [DispatcherQueue.GetForCurrentThread](https://docs.microsoft.com/uwp/api/Windows.System.DispatcherQueue.GetForCurrentThread)
+to see if this is the case.
+
+See also [CreateDispatcherQueueController](https://docs.microsoft.com/en-us/windows/win32/api/dispatcherqueue/nf-dispatcherqueue-createdispatcherqueuecontroller)
+
+
+# Examples
+
+This example shows a thread ensuring it has a DispatcherQueue, setting a Synchronization
+context onto the thread that uses that queue, and then goes into a message pump.
+
+```cs
+static void Run()
+{
+    // Ensure we have a DispatcherQueue on this thread. 

```

Comment:
Please update comment:
// Note - this pattern is only necessary if you need to create a DispatcherQueue on a thread (ex as part
// of initialization).  For the vast majority of scenarios, a hosting framework has already created one
// on your behalf and so you should simply call DispatcherQueue.GetForCurrentThread() to retrieve it.

Reply:
@scott-moore-ms - look OK?

Reply:
LGTM